lib/Pool.js

```diff
@@ -339,6 +372,23 @@ class Pool {
     return new Promise((resolve, reject) => {
       odbc.connect(connectionConfig, (error, nativeConnection) => {
         if (error) {
+          // When there is a connection error, emit a "connectionError" event
+          // with the error that is then either handled by any waiting work, or // is silently swallowed if no work is waiting. This is so that a bad
+          // attempt at connecting doesn't throw an uncatchable error when there
+          // is no user work, crashing the entire application.
+          //
+          // Catchable errors occur:
+          // 1. When a pool is initialized and there was a problem with the
+          // first connection (no network, bad connection string, etc.)
+          // 2. When a user is attempting to do work and there are no active
+          // connections, the pool will attempt to connect, emit the error, and
+          // send it to the user work that was waiting (either as an error
+          // parameter in the user's callback, or as a thrown error that the
+          // user has to catch.)
+          this.pool.connectionEmitter.emit('connectionError', error);
+
+          // This reject is swallowed in all cases, and is simply to end the
```

Comment:
I'd recommend mentioning where it is swallowed, so someone viewing it can follow the flow. Something like

```suggestion
          // This reject is swallowed in all cases inside the dequeue function, and is simply to end the
```